src/OfflineSearch.xml

```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="utf-8"?>
+<widget id="com.mendix.widget.custom.OfflineSearch.OfflineSearch" offlineCapable="true" needsEntityContext="false" xmlns="http://www.mendix.com/widget/1.0/">
+    <name>Offline Search</name>
+    <description>Search Mendix List views, Template Grids and Data Grids using a single-field search</description>
+    <icon />
+
+    <properties>
+        <property key="targetGridName" type="string" required="false" defaultValue="">
+            <caption>List view name</caption>
+            <category>Data source</category>
+            <description>The name of the grid on this page</description>
+        </property>
+        <property key="searchMethod" type="enumeration" defaultValue="contains">
+            <caption>Search method</caption>
+            <category>Data source</category>
+            <description>Search method to use</description>
+            <enumerationValues>
+                <enumerationValue key="equals">Equals</enumerationValue>
+                <enumerationValue key="contains">Contains</enumerationValue>
+            </enumerationValues>
+        </property>
+        <property key="searchEntity" type="string" required="false">
```

Comment:
we use it and its optional. `searchEntity` is used to search through related entity to the `Entity` on the listview